app/models/manageiq/providers/inventory/persister/builder/cloud_manager.rb

```diff
@@ -165,17 +165,21 @@ def vm_and_miq_template_ancestry_save_block
 
           # Fetch IDs of all vms and genealogy_parents, only if genealogy_parent is present
           vms_genealogy_parents = vms_inventory_collection.data.each_with_object({}) do |x, obj|
-            unless x.data[:genealogy_parent].nil?
-              genealogy_parent_id = x.data[:genealogy_parent].load.try(:id)
-              obj[x.id] = genealogy_parent_id if genealogy_parent_id
-            end
+            next if x.data[:genealogy_parent].nil?
+
+            genealogy_parent_id = x.data[:genealogy_parent].load.try(:id)
+            next if genealogy_parent_id.nil? || genealogy_parent_id == x.id
```

Comment:
I feel like this should emit a warning, since it seems to indicate a problem with either the parser or the raw data from the provider.

Reply:
:+1: yeah that's easy to do